nixos/modules/services/security/bitwarden-directory-connector.nix

```diff
@@ -0,0 +1,336 @@
+          type = types.str;
+          description = lib.mdDoc "Attribute that has a users username.";
+          example = "accountName";
+        };
+        suffix = mkOption {
+          type = types.str;
+          description = lib.mdDoc "Suffix for the email, normally @example.com.";
+          example = "@example.com";
+        };
+      };
+
+      users = {
+        enable = mkOption {
+          type = types.bool;
+          default = false;
+          description = lib.mdDoc "Sync users.";
+        };
+        path = mkOption {
+          type = types.str;
+          description = lib.mdDoc "User directory, relative to root.";
+          example = "ou=users";
+        };
+        objectClass = mkOption {
+          type = types.str;
+          description = lib.mdDoc "A class that users will have.";
+          example = "inetOrgPerson";
+        };
+        emailAttribute = mkOption {
+          type = types.str;
+          description = lib.mdDoc "Attribute for a users email.";
+          example = "mail";
+        };
+        filter = mkOption {
+          type = types.str;
+          description = lib.mdDoc "Filter for users.";
+          example = "(memberOf=cn=sales,ou=groups,dc=example,dc=com)";
+        };
+      };
+      groups = {
+        enable = mkOption {
+          type = types.bool;
+          default = false;
+          description = lib.mdDoc "Sync groups.";
+        };
+        path = mkOption {
+          type = types.str;
+          description = lib.mdDoc "Group directory, relative to root.";
+          example = "ou=groups";
+        };
+        objectClass = mkOption {
+          type = types.str;
+          description = lib.mdDoc "A class that groups will have.";
+          example = "groupOfNames";
+        };
+        nameAttribute = mkOption {
+          type = types.str;
+          description = lib.mdDoc "Attribute for a name of group.";
+          example = "cn";
+        };
+        filter = mkOption {
+          type = types.str;
+          description = lib.mdDoc "Filter for groups.";
+          example = "(cn=sales)";
+        };
+      };
+    };
+
+    env = {
+      ldap = mkOption rec {
+        type = types.str;
+        description = "Auth for the LDAP, has value defined in {option}`ldap.pw_env";
+      };
+      bitwarden = mkOption rec {
+        type = types.str;
+        description = "Auth for Bitwarden, has BW_CLIENTID and BW_CLIENTSECRET";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.groups."${cfg.user}" = {};
+
+    users.users."${cfg.user}" = {
+      createHome = true;
+      isSystemUser = true;
+      home = "${cfg.directory}";
+      group = "${cfg.user}";
+      homeMode = "711";
+    };
+
+    systemd = {
+      timers.bitwarden_directory_connector = {
+        description = "Timer for ${cfg.user}";
+        wantedBy = ["timers.target"];
+        partOf = ["${cfg.user}.service"];
+        timerConfig = {
+          OnCalendar = cfg.sync.interval;
```

Comment:
Does this overwrite the "interval: 5" above?

Reply:
Yes, kindof, as I said in my other comment the other value hsa no use in the cli version, this option is how the user sets how often/when to sync.